### PR TITLE
[fixes] issues from #929

### PR DESCRIPTION
**Pull request by kvedala**
#### Description of Change

* Comments made in #929  fixed
* added function overload
* included function in namespace `sorting` for consistency

#### Checklist

- [x] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by kvedala** [comment]
@ayaankhan98 you can merge this if all looks good. thanks

File: sorting/insertion_sort.cpp [hunk 1]
```diff
@@ -4,67 +4,61 @@
...
- * <pre>
- * 1 - easy to implement
- * 2 - For small set of data it is quite efficient
- * 3 - More efficient that other Quadratic complexity algorithms like
- *     Selection sort or bubble sort.
- * 4 - It's stable that is it does not change the relative order of
- *     elements with equal keys
- * 5 - Works on hand means it can sort the array or list as it receives.
- * </pre>
+ * 1. easy to implement
```
Comment (ayaankhan98) [inline sorting/insertion_sort.cpp:12]: it's not a big issue, but it's good to replace
```suggestion
 * 1. Easy to implement
```

File: sorting/insertion_sort.cpp [hunk 2]
```diff
@@ -4,67 +4,61 @@
...
- * 2 3 4 5 1
- *
- * we does not do anything with 5 and move on to the next iteration and select
- * 1 which is misplaced and place it at correct position. Thus, we have
- *
- * 1 2 3 4 5
- * </pre>
- *
+ * \f{bmatrix}{2 &3 &4 &5 &1\f}
+ * 4. we does not do anything with 5 and move on to the next iteration and
```
Comment (Panquesito7) [inline sorting/insertion_sort.cpp:37]:
```suggestion
 * 4. We do not do anything with 5 and move on to the next iteration and
```

Comment (kvedala) [reply sorting/insertion_sort.cpp:37]: > 

```suggestion
 * 4. we do not do anything with 5 and move on to the next iteration and
```

File: sorting/insertion_sort.cpp [hunk 3]
```diff
@@ -34,7 +34,7 @@
  * we place it at the correct sorted position thus the array in this iteration
  * becomes
  * \f{bmatrix}{2 &3 &4 &5 &1\f}
- * 4. we does not do anything with 5 and move on to the next iteration and
+ * 4. we do not do anything with 5 and move on to the next iteration and
```
Comment (kvedala) [inline sorting/insertion_sort.cpp:37]:
```suggestion
 * 4. We do not do anything with 5 and move on to the next iteration and
```

**Comment by kvedala** [comment]
all good? lets merge

**Pull request merged**